lib/galaxy/tools/__init__.py

```diff
@@ -3563,6 +3563,65 @@ def produce_outputs(self, trans, out_data, output_collections, incoming, history
         )
 
 
+class SynchronizeTool(DatabaseOperationTool):
+    tool_type = "synchronize_list"
+    require_terminal_states = False
+    require_dataset_ok = False
+
+    def produce_outputs(self, trans, out_data, output_collections, incoming, history, **kwds):
+        # Get the 2 input collections
+        hdca1 = incoming["input1"]
+        hdca2 = incoming["input2"]
+        # Get the elements of both collections
+        elements1 = hdca1.collection.elements
+        elements2 = hdca2.collection.elements
+        # Put elements in dictionary with identifiers:
+        old_elements1_dict = {}
+        for element in elements1:
+            old_elements1_dict[element.element_identifier] = element
+        old_elements2_dict = {}
+        for element in elements2:
+            old_elements2_dict[element.element_identifier] = element
+        # Get the list of final identifiers
+        final_sorted_identifiers = [element.element_identifier for element in elements1
+                                    if element.element_identifier in old_elements2_dict]
+        # Raise Exception if it is empty
+        if len(final_sorted_identifiers) == 0:
+            hdca1_history_name = f"{hdca1.hid}: {hdca1.name}"
+            hdca2_history_name = f"{hdca2.hid}: {hdca2.name}"
+            message = f"The two collections '{hdca1_history_name}' and '{hdca2_history_name}'" \
+                " have no identifier in common. It is not possible to synchronize them."
+            raise Exception(message)
```

Comment:
I think that's not necessarily an error, you may just legitimately have an empty collection

Reply:
OK